gslib/utils/copy_helper.py

```diff
@@ -335,7 +335,8 @@ def _PerformParallelUploadFileToObject(cls, args, thread_state=None):
     dst_object_metadata = apitools_messages.Object(
         name=args.dst_url.object_name,
         bucket=args.dst_url.bucket_name,
-        contentType=args.content_type)
+        contentType=args.content_type,
+        storageClass=args.storage_class)
```

Comment:
Storage class isn't a consideration for sliced downloads because the file system destination doesn't have a concept of storage classes.